### ISSUE TITLE
System.ArgumentNullException: Value cannot be null. (Parameter 'source') when using {0} in property set template

### ISSUE THREAD

**Issue by br0esel** [yanghuan/CSharp.lua#322]
I tried to change the generated code of System.Console.BackgroundColor set 

I added this to System.xml

```xml
 <property name="BackgroundColor">
   <set Template="bla{0}"/>
 <property>
``` 
which results in the following exception
```
3>CSharpLua.BugErrorException: SourceLocation(C:\Users\denni\source\repos\Opencomputers.CSharp\ConsoleApp2\Program.cs@45:13)"Console.BackgroundColor = ConsoleColor.DarkRed;": Compiler has a bug, thanks to commit issue at https://github.com/yanghuan/CSharp.lua/issue
3> ---> System.ArgumentNullException: Value cannot be null. (Parameter 'source')
3>   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
3>   at System.Linq.Enumerable.ElementAtOrDefault[TSource](IEnumerable`1 source, Int32 index)
3>   at CSharpLua.LuaSyntaxNodeTransform.InternalBuildCodeTemplateExpression(String codeTemplate, ExpressionSyntax targetExpression, IEnumerable`1 arguments, IList`1 typeArguments, LuaIdentifierNameSyntax memberBindingIdentifier) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.Helper.cs:line 352
3>   at CSharpLua.LuaSyntaxNodeTransform.BuildCodeTemplateExpression(String codeTemplate, ExpressionSyntax targetExpression) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.Helper.cs:line 280
3>   at CSharpLua.LuaSyntaxNodeTransform.CheckMemberAccessCodeTemplate(ISymbol symbol, MemberAccessExpressionSyntax node) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 2480
3>   at CSharpLua.LuaSyntaxNodeTransform.InternalVisitMemberAccessExpression(ISymbol symbol, MemberAccessExpressionSyntax node) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 2491
3>   at CSharpLua.LuaSyntaxNodeTransform.VisitMemberAccessExpression(MemberAccessExpressionSyntax node) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 2609
3>   at Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
3>   at CSharpLua.Utility.Accept[T](CSharpSyntaxNode node, LuaSyntaxNodeTransform transform) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\Utils.cs:line 1283
3>   at CSharpLua.Utility.AcceptExpression(CSharpSyntaxNode node, LuaSyntaxNodeTransform transform) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\Utils.cs:line 1287
3>   at CSharpLua.LuaSyntaxNodeTransform.BuildLuaAssignmentExpression(AssignmentExpressionSyntax node, ExpressionSyntax leftNode, ExpressionSyntax rightNode, SyntaxKind kind) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 1784
3>   at CSharpLua.LuaSyntaxNodeTransform.InternalVisitAssignmentExpression(AssignmentExpressionSyntax node) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 1940
3>   at CSharpLua.LuaSyntaxNodeTransform.VisitAssignmentExpression(AssignmentExpressionSyntax node) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 1997
3>   at Microsoft.CodeAnalysis.CSharp.Syntax.AssignmentExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
3>   at CSharpLua.Utility.Accept[T](CSharpSyntaxNode node, LuaSyntaxNodeTransform transform) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\Utils.cs:line 1283
3>   at CSharpLua.Utility.AcceptExpression(CSharpSyntaxNode node, LuaSyntaxNodeTransform transform) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\Utils.cs:line 1287
3>   at CSharpLua.LuaSyntaxNodeTransform.VisitExpressionStatement(ExpressionStatementSyntax node) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 1489
3>   at Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionStatementSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
3>   at CSharpLua.LuaSyntaxNodeTransform.BlockCommonNode.Visit(LuaSyntaxNodeTransform transfor) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxNodeTransform.cs:line 1292
3>   --- End of inner exception stack trace ---
3>   at CSharpLua.LuaSyntaxGenerator.Create() in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxGenerator.cs:line 220
3>   at CSharpLua.LuaSyntaxGenerator.Generate(String outFolder) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\LuaSyntaxGenerator.cs:line 246
3>   at CSharpLua.Compiler.Compile() in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua\Compiler.cs:line 120
3>   at CSharpLua.Program.Main(String[] args) in C:\Users\denni\source\repos\Opencomputers.CSharp\CSharp.lua.Launcher\Program.cs:line 93
```

**Comment by br0esel** [comment]
I think I used it wrong. My intention was to overwrite the property's getter and setter methods with my own template in system.xml, but I don't know how to do that.

**Comment by yanghuan** [comment]
fix by 2b2ac839babf1c2483a572eaf677e47b83d437b9
